processor/src/main/java/com/linecorp/decaton/processor/runtime/TaskExtractor.java

```diff
@@ -17,16 +17,16 @@
 package com.linecorp.decaton.processor.runtime;
 
 /**
- * An interface for classes extracting {@link DecatonTask} from given bytes.
+ * An interface for classes extracting {@link DecatonTask} from given record.
  * @param <T> type of task.
  */
 public interface TaskExtractor<T> {
     /**
      * Extract object of type {@link DecatonTask} from given bytes.
-     * @param bytes raw message bytes.
+     * @param record {@link ConsumedRecord} to extract task from.
      * @return object of type {@link DecatonTask}.
-     * @throws RuntimeException this method can throw arbitrary {@link RuntimeException} if given bytes is invalid.
+     * @throws RuntimeException this method can throw arbitrary {@link RuntimeException} if given record is invalid.
      * If the method throws an exception, the task will be discarded and processor continues to process subsequent tasks.
      */
-    DecatonTask<T> extract(byte[] bytes);
+    DecatonTask<T> extract(ConsumedRecord record);
```

Comment:
So we're effectively exposing kafka specific API (Headers) to users through this?

Reply:
Yes.
In fact, we already expose Headers through `ProcessingContext#headers`.

We might extend Decaton to support other type of mq in the future though, then we need to make change to `ProcessingContext#headers` too so I think we're ok to expose headers for now.

Of course, we should be cautious to avoid making Decaton couple to specific mq as much as possible though.